code/datums/components/mood.dm

```diff
@@ -220,31 +210,8 @@
 	master.crit_threshold = (master.crit_threshold - insanity_effect) + newval
 	insanity_effect = newval
 
-/datum/component/mood/proc/DecreaseSanity(datum/source, amount, minimum = SANITY_INSANE)
-	if(sanity < minimum) //This might make KevinZ stop fucking pinging me.
-		IncreaseSanity(src, 0.5)
-	else
-		sanity = max(minimum, sanity - amount)
-		if(sanity < SANITY_UNSTABLE)
-			if(sanity < SANITY_CRAZY)
-				insanity_effect = (MAJOR_INSANITY_PEN)
-			else
-				insanity_effect = (MINOR_INSANITY_PEN)
-
-/datum/component/mood/proc/IncreaseSanity(datum/source, amount, maximum = SANITY_NEUTRAL)
-	// Disturbed stops you from getting any more sane - I'm just gonna bung this in here
-	var/mob/living/owner = parent
-	if(HAS_TRAIT(owner, TRAIT_UNSTABLE))
-		return
-	if(sanity > maximum)
-		DecreaseSanity(src, 0.5) //Removes some sanity to go back to our current limit.
-	else
-		sanity = min(maximum, sanity + amount)
-		if(sanity > SANITY_CRAZY)
-			if(sanity > SANITY_UNSTABLE)
-				insanity_effect = 0
-			else
-				insanity_effect = MINOR_INSANITY_PEN
+/datum/component/mood/proc/modify_sanity(datum/source, amount, minimum = -INFINITY, maximum = INFINITY)
+	setSanity(sanity + amount, minimum, maximum)
```

Comment:
Yes, it's pretty stupid, but I'm leaving any further mood refactor and fix for future separate PRs.